Automattic/jetpack
# Instant Search: Add support for hiding search sort selector

Split out from #15727.

Fixes #15735 

#### Changes proposed in this Pull Request:
* Adds a site-wide option to hide the sort selector in the Jetpack Search overlay.
* Removes the `Show search box` option from the Jetpack Search widget when instant search is enabled.
* Removes the `Show sort selection dropdown` option from the Jetpack Search widget when instant search is enabled. 
* Rewords various Jetpack Search customizer section descriptions for brevity and consistency.
* Adds a custom `Label_Control` class to be used in the Jetpack Search customizer section.
* Refactors the bridging logic between the Customizer and the Instant Search app to avoid duplicating code.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Yes, it adds the ability to hide the sort selector in the Jetpack Search overlay. The search overlay is only available to customers with the Search product plan.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Apply this change to a site with Jetpack Instant Search enabled.

2. Add a Jetpack Search widget to your site's footer or sidebar in the customizer.

3. Ensure that neither `Show search box` nor `Show sort selection dropdown` checkboxes are available. 

> <img width="292" alt="Screen Shot 2020-05-28 at 5 33 35 PM" src="https://user-images.githubusercontent.com/4044428/83204576-6507a980-a109-11ea-894c-9875d9db2a00.png">

4. Ensure that the widget title, post types checkboxes, default sort order, and filter controls work as expected. These options should not be affected by this change.

5. Navigate to the Jetpack Search section of the customizer. 

6. Ensure there's a group of checkboxes titled `Additional Jetpack Search Settings`. 

> <img width="292" alt="Screen Shot 2020-05-28 at 5 35 55 PM" src="https://user-images.githubusercontent.com/4044428/83204699-b44dda00-a109-11ea-9b3e-d32cca6ac4a5.png">

7. Try toggling the newly added `Show Sort Selector` checkbox. Verify that unchecking the checkbox hides the sort selector in the search overlay without refreshing the page.

> <img width="686" alt="Screen Shot 2020-05-11 at 3 13 39 PM" src="https://user-images.githubusercontent.com/4044428/81612412-036dde00-939a-11ea-9398-9915a1721762.png">
> <img width="686" alt="Screen Shot 2020-05-11 at 3 14 36 PM" src="https://user-images.githubusercontent.com/4044428/81612478-239d9d00-939a-11ea-86eb-a27c069dd344.png">

8. Try disabling instant search via `/wp-admin/admin.php?page=jetpack#/performance` and repeat step 3. Both `Show search box` and `Show sort selection dropdown` checkboxes should be available again.

#### Proposed changelog entry for your changes:
* Adds a Customizer control for hiding the sort selector in the Jetpack Search overlay.
